### PR TITLE
1.25 fixes

### PR DESCRIPTION
**Pull request by FilipLeitner**
- clustering
- wms layer max resolution
- Add support for .wmc compositions
- Add compositions error dialog
- sublayers styling react to resolution changes

### CODE REVIEW

File: components/layermanager/layermanager-metadata.service.js [hunk 1]
```diff
@@ -183,7 +183,9 @@ export default function (
               layer.getProperties()
             );
             console.log('final', maxScale);
-            layer.set('maxResolution', me.roundToHundrets(maxScale));
+            if(maxScale){
```
Comment (raitisbe) [inline components/layermanager/layermanager-metadata.service.js:186]: Hundreds :)

File: components/layermanager/layermanager-metadata.service.js [hunk 2]
```diff
@@ -62,6 +62,33 @@ export default function (
...
+   * @returns {number}
+   */
+  me.roundToHundrets = function (num) {
+    return Math.ceil(num / 100) * 100;
+  };
+
+  me.searchForScaleDenominator = function (layer) {
+    let maxScale = layer.MaxScaleDenominator ? layer.MaxScaleDenominator : null;
+    const layers = layer.Layer;
+    for (const index in layers) {
```
Comment (raitisbe) [inline components/layermanager/layermanager-metadata.service.js:79]: for of  better maybe?

File: components/layermanager/layermanager-metadata.service.js [hunk 3]
```diff
@@ -143,7 +170,23 @@ export default function (
...
+          $timeout(() => {
+            if (layer.get('MaxScaleDenominator')) {
+              layer.set(
+                'maxResolution',
+                me.roundToHundrets(layer.get('MaxScaleDenominator'))
+              );
+              return;
+            }
+            const maxScale = me.searchForScaleDenominator(
+              layer.getProperties()
```
Comment (raitisbe) [inline components/layermanager/layermanager-metadata.service.js:183]: Why not just layer without getProperties?

File: components/layermanager/layermanager.service.js [hunk 4]
```diff
@@ -713,9 +711,12 @@ export default function (
    * @function isLayerInResolutionInterval
    * @memberOf HsLayermanagerService
    * @param {Ol.layer} lyr Selected layer
-   * @description Test if layer (WMS) resolution is within map resolution interval
+   * @description Test if layer (WMS) resolution is NOT within map resolution interval. True = out of the interval
```
Comment (raitisbe) [inline components/layermanager/layermanager.service.js:714]: I think it should have been "Map resolution is within layers resolution interval minResolution <= current <= maxResolution"

Comment (raitisbe) [reply components/layermanager/layermanager.service.js:714]: the condition
```
lyr.getMinResolution() >= cur_res && cur_res <= lyr.getMaxResolution()
```
Seems to be the correct one.

File: components/layermanager/layermanager-metadata.service.js (hunk 3 above)
Comment (FilipLeitner) [reply components/layermanager/layermanager-metadata.service.js:183]: Not sure why I went this way. ty

Comment (FilipLeitner) [reply components/layermanager/layermanager-metadata.service.js:183]: searchForScaleDenominator is recursive while sublayers are not instances of ol.layer so they dont have getProperties method

File: components/layermanager/layermanager.service.js (hunk 4 above)
Comment (FilipLeitner) [reply components/layermanager/layermanager.service.js:714]: This ones bit confusing. The logic while using the return vaule is twisted:
`grayed: me.isLayerInResolutionInterval(layer)` .. should not be grayed if in resolution

Comment (FilipLeitner) [reply components/layermanager/layermanager.service.js]: Made some changes, seems to work fine now.
2a93397e3017698934abee272a92e293aee594a9

File: components/layermanager/layermanager.service.js [hunk 5]
```diff
@@ -730,6 +733,7 @@ export default function (
       cur_res = HsMapService.map.getView().getResolution();
     }
     me.currentResolution = cur_res;
+    console.log(cur_res, lyr.getMaxResolution());
```
Comment (raitisbe) [inline components/layermanager/layermanager.service.js:736]: Probably a leftover?

Comment (FilipLeitner) [reply components/layermanager/layermanager.service.js:736]: O right, thanks. Will resolve with backporting of https://github.com/hslayers/hslayers-ng/pull/1206

**Pull request merged**